internal/render/blocks.go

```
@@ -0,0 +1,456 @@
+
+// AlertFallbackText returns the plain-text fallback for an alert.
+func AlertFallbackText(level AlertLevel, message string) string {
+	return fmt.Sprintf("[%s] %s", strings.ToUpper(string(level)), truncateStr(message, 100))
+}
+
+// ---------------------------------------------------------------------------
+// Helpers
+// ---------------------------------------------------------------------------
+
+// hostnameFromURL extracts just the hostname from a URL string.
+func hostnameFromURL(rawURL string) string {
```

golangci-lint (2.12.2)

[error] func hostnameFromURL is unused

(unused)